frugal field type not provided will cause panic

## Produce

If I simply run the examples in the `Serialization and deserialization on a customized Go struct` section in README.md, it will get an error message:

```
panic: frugal: cannot measure encoded size: invalid tag for field main.MyStruct.Msg

goroutine 1 [running]:
github.com/cloudwego/frugal/internal/binary/encoder.EncodedSize({0xf714e0, 0xc00010f620})
        /home/yunwei/coding/frugal/internal/binary/encoder/encoder.go:69 +0x8e
github.com/cloudwego/frugal.EncodedSize(...)
        /home/yunwei/coding/frugal/frugal.go:27
main.main()
        /home/yunwei/coding/frugal/cmd/main.go:20 +0xa9
exit status 2
```

the test code is like this, which is copied from README:

```go
package main

import (
    "github.com/cloudwego/frugal"
)

type MyStruct struct {
    Msg     string  `frugal:"1,default"`
    Code    int64   `frugal:"2,default"`
    Numbers []int64 `frugal:"3,default,list<i64>"`
}

func main() {
    ms := &MyStruct{
        Msg: "my message",
        Code: 1024,
        Numbers: []int64{0, 1, 2, 3, 4},
    }

    buf := make([]byte, frugal.EncodedSize(ms))
    frugal.EncodeObject(buf, nil, ms)

    got := &MyStruct{}
    frugal.DecodeObject(buf, got)
}
```

However, if I change the struct to:

```go
type MyStruct struct {
    Msg     string  `frugal:"1,default,string"`
    Code    int64   `frugal:"2,default,i64"`
    Numbers []int64 `frugal:"3,default,list<i64>"`
}
```

Then the example can be run successfully.

## Question

The README said:

> Frugal tag is like `frugal:"1,default,string"`, `1` is field ID, `default` is field requiredness, `string` is field type. Field ID and requiredness is always required, but field type is only required for `list`, `set` and `enum`.

while the code in the `internal/binary/defs/resolver.go` gives that:

```go
        /* must have 3 fields: ID, Requiredness, Type */
        if ft = strings.Split(tv, ","); len(ft) != 3 {
            return nil, fmt.Errorf("invalid tag for field %s.%s", vt, sf.Name)
        }
```

Is the `field type` optional tag not support currently, or the tag should be provided each time in Frugal? 

By the way, another minor fix for README example is in PR #7 .

--- a/internal/binary/defs/resolver.go
+++ b/internal/binary/defs/resolver.go
@@ -151,10 +151,13 @@ func doResolveFields(vt reflect.Type) ([]Field, error) {
             continue
         }
 
-        /* must have at least 3 fields: ID, Requiredness, Type */
-        if ft = strings.Split(tv, ","); len(ft) < 3 {
+        /* must have at least 2 fields: ID and Requiredness; Type and other options are optional */
+        if ft = strings.Split(tv, ","); len(ft) < 2 {
             return nil, fmt.Errorf("invalid tag for field %s.%s", vt, sf.Name)
-        }
+        } else if len(ft) == 2 {
+	    /* append ft with an empty Type field */
+            ft = append(ft, "")
+	}
 
         /* parse the field index */
         if id, err = strconv.ParseUint(strings.TrimSpace(ft[0]), 10, 16); err != nil {

--- a/internal/binary/defs/resolver_test.go
+++ b/internal/binary/defs/resolver_test.go
@@ -25,8 +25,11 @@ import (
 )
 
 type NoCopyStringFields struct {
-    NormalString string `frugal:"1,default,string"`
-    NoCopyString string `frugal:"2,default,string,nocopy"`
+    NormalString         string `frugal:"1,default,string"`
+    NoCopyString         string `frugal:"2,default,string,nocopy"`
+    TypelessString       string `frugal:"3,default"`
+    TypelessString2      string `frugal:"4,default,"`
+    NoCopyTypelessString string `frugal:"5,default,,nocopy"`
 }
 
 func TestResolver_StringOptions(t *testing.T) {

